gmail/src/models/task.ts

```diff
@@ -0,0 +1,60 @@
+import { postJsonRpc } from "../utils/http";
+import { URLS } from "../const";
+import { State } from "../models/state";
+
+/**
+ * Represent a "project.task" record.
+ */
+export class Task {
+    id: number;
+    name: string;
+    projectName: string;
+
+    /**
+     * Unserialize the task object (reverse JSON.stringify).
+     */
+    static fromJson(values: any): Task {
+        const task = new Task();
+        task.id = values.id;
+        task.name = values.name;
+        task.projectName = values.projectName;
+        return task;
+    }
+
+    /**
+     * Parse the dictionary return by the Odoo endpoint.
+     */
+    static fromOdooResponse(values: any): Task {
+        const task = new Task();
+        task.id = values.task_id;
+        task.name = values.name;
+        task.projectName = values.project_name;
+        return task;
+    }
+
+    /**
+     * Make a RPC call to the Odoo database to create a task
+     * and return the ID of the newly created record.
+     */
+    static createTask(partnerId: number, projectId: number, emailBody: string, emailSubject: string): Task {
+        const url = State.odooServerUrl + URLS.CREATE_TASK;
+        const accessToken = State.accessToken;
+
+        const response = postJsonRpc(
+            url,
+            { email_body: emailBody, email_subject: emailSubject, partner_id: partnerId, project_id: projectId },
```

Comment:
Args order on route is ``email_subject, email_body, project_id, partner_id`` . As those are not named parameter, how does it behave here, especially that partner / project are inverted ?

Reply:
Even if it's not "named" on the python side, the "Json Dispatch" on the Odoo side will put the value in the right parameter. But for consistency, I can use the same order :)